src/components/Checkbox/Checkbox.Props.ts

```diff
@@ -1,31 +1,56 @@
 import * as React from 'react';
 import { Checkbox } from './Checkbox';
 
+/**
+ * Checkbox class interface.
+ */
+export interface ICheckbox {
+  /** Gets the current checked state. */
+  checked: boolean;
+
+  /** Sets focus to the checkbox. */
+  focus(): void;
+}
+
+/**
+ * Checkbox properties.
+ */
 export interface ICheckboxProps extends React.Props<Checkbox> {
   /**
-   * Label to display next to the checkbox.
+   * Additional class name to provide on the root element, in addition to the ms-Checkbox class.
    */
-  text?: string;
+  className?: string;
 
   /**
-   * Whether the checkbox is checked or not.
-   * @defaultvalue false
+   * Checked state. Mutually exclusive to "defaultChecked". Use this if you control the checked state at a higher
+   * level and plan to pass in the correct value based on handling onChange events and re-rendering.
    */
-  isChecked?: boolean;
+  checked?: boolean;
 
   /**
-   * Whether the checkbox is enabled or not.
-   * @defaultvalue true
+   * Default checked state. Mututally exclusive to "checked". Use this if you want an uncontrolled component, and
+   * want the Checkbox instance to maintain its own state.
    */
-  isEnabled?: boolean;
+  defaultChecked?: boolean;
+
+  /**
+   * Label to display next to the checkbox.
+   */
+  label?: string;
+
+  /**
+   * Disabled state of the checkbox.
+   */
+  disabled?: boolean;
 
   /**
    * Callback that is called when the checked value has changed.
    */
-  onChanged?: (isChecked: boolean) => void;
+  onChange?: (ev?: React.FormEvent, isChecked?: boolean) => void;
 
   /**
-   * Optional class name to stamp on the root of the component.
+   * Input props that will be mixed into the input element, before other props are applied. This allows you to extend
+   * the input element with additional attributes, such as data-automation-id needed for automation.
```

Comment:
Should mention how it interacts iwth things like disabled and onChange if specified in both places